Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp

```diff
@@ -0,0 +1,187 @@
+        return API::Data::create(generatedBackgroundContent().utf8().span());
+
+    auto resourceURL = resourceFileURLForPath(path);
+    if (!resourceURL.isEmpty()) {
+        if (suppressErrors == SuppressNotFoundErrors::No)
+            outError = createError(Error::ResourceNotFound, WEB_UI_FORMAT_STRING("Unable to find \"%s\" in the extension’s resources. It is an invalid path.", "WKWebExtensionErrorResourceNotFound description with invalid file path", path.utf8().data()));
+        return nullptr;
+    }
+
+    auto rawData = FileSystem::readEntireFile((resourceURL).fileSystemPath());
+    if (!rawData.has_value()) {
+        if (suppressErrors == SuppressNotFoundErrors::No)
+            outError = createError(Error::ResourceNotFound, WEB_UI_FORMAT_STRING("Unable to find \"%s\" in the extension’s resources.", "WKWebExtensionErrorResourceNotFound description with file name", path.utf8().data()));
+        return nullptr;
+    }
+
+    auto data = API::Data::create(*rawData);
+
+    if (cacheResult == CacheResult::Yes)
+        m_resources.set(path, data);
+
+    return data;
+}
+
+void WebExtension::recordError(Ref<API::Error> error)
+{
+    RELEASE_LOG_ERROR(Extensions, "Error recorded: %s", error->platformError());
+
+    // Only the first occurrence of each error is recorded in the array. This prevents duplicate errors,
+    // such as repeated "resource not found" errors, from being included multiple times.
+    if (m_errors.contains(error))
+        return;
+
+    m_errors.append(error);
+}
+
+RefPtr<WebCore::Icon> WebExtension::iconForPath(const String& path, RefPtr<API::Error>& error, WebCore::FloatSize sizeForResizing)
+{
+    ASSERT(path);
+
+    auto imageData = resourceDataForPath(path, error);
+    if (imageData->span().empty())
+        return nullptr;
+
+    auto gimageBytes = adoptGRef(g_bytes_new(imageData->span().data(), imageData->size()));
+
+    if (!sizeForResizing.isZero()) {
+        auto loader = adoptGRef(gdk_pixbuf_loader_new());
+        gdk_pixbuf_loader_write_bytes(loader.get(), gimageBytes.get(), nullptr);
+        gdk_pixbuf_loader_close(loader.get(), nullptr);
+        auto pixbuf = adoptGRef(gdk_pixbuf_copy(gdk_pixbuf_loader_get_pixbuf(loader.get())));
+        if (!pixbuf)
+            return nullptr;
+
+        // Proportionally scale the size
+        auto originalWidth = gdk_pixbuf_get_width(pixbuf.get());
+        auto originalHeight = gdk_pixbuf_get_height(pixbuf.get());
+        auto aspectWidth = sizeForResizing.width() / originalWidth;
+        auto aspectHeight = sizeForResizing.height() / originalHeight;
+        auto aspectRatio = std::min(aspectWidth, aspectHeight);
+
+        gdk_pixbuf_scale_simple(pixbuf.get(), originalWidth * aspectRatio, originalHeight * aspectRatio, GDK_INTERP_BILINEAR);
+
+        gchar* buffer;
+        gsize* bufferSize;
+        gdk_pixbuf_save_to_buffer(pixbuf.get(), &buffer, bufferSize, "png"_s, NULL, NULL);
```

Comment:
```suggestion
        gdk_pixbuf_save_to_buffer(pixbuf.get(), &buffer, bufferSize, "png"_s, nullptr, nullptr);
```

Are you sure it's OK to ignore errors here? I almost never ignore the error parameter since it makes debugging problems so much harder.